kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -196,15 +218,39 @@ impl<'tcx> GotocCtx<'tcx> {
             ($f:ident) => {{
                 let a = fargs.remove(0);
                 let b = fargs.remove(0);
+                let op_type = a.typ().clone();
                 let res = a.$f(b);
+                // add to symbol table
+                let res_type = arithmetic_overflow_result_type(op_type.clone());
+                let struct_tag =
+                    self.ensure_struct(res_type.tag().unwrap(), res_type.tag().unwrap(), |_, _| {
+                        res_type.components().unwrap().clone()
+                    });
+                assert_eq!(*res.typ(), struct_tag);
+
+                // store the result in a temporary variable
+                let (var, decl) = self.decl_temp_variable(struct_tag, Some(res), loc);
                 let check = self.codegen_assert(
-                    res.overflowed.not(),
+                    var.clone()
+                        .member(ARITH_OVERFLOW_OVERFLOWED_FIELD, &self.symbol_table)
+                        .cast_to(Type::c_bool())
+                        .not(),
```

Comment:
Were you going to split this off?

Reply:
This case wasn't difficult to handle. The other cases, e.g. in `count_in_bytes`, `codegen_ptr_offset_from_expr`, and `codegen_ptr_offset_from_unsigned` are a bit more challenging, and I'm splitting those off to a later PR.